src/org/mozilla/javascript/NativeJavaObject.java

```diff
@@ -695,8 +695,7 @@ else if (value instanceof Wrapper) {
                 reportConversionError(value, type);
             }
             else if (type.isInterface() && (value instanceof NativeObject
-                    || value instanceof NativeFunction
-                    || value instanceof ArrowFunction)) {
+                    || value instanceof Callable)) {
```

Comment:
I'm pretty sure that this is OK as long as value is also a ScriptableObject -- see the cast below and the comment in "createInterfaceAdapter." It's possible for something to be Callable but not a ScriptableObject, and in that case this change will result in a ClassCastException.

Reply:
So sorry, my comment wasn't clear -- I think it's fine to check "instanceof Callable" here and I like that it fixes a number of other problems, but we also need to check "instanceof ScriptableObject" in case people have written code that is Callable but does not derive from ScriptableObject.

Reply:
7fdd75e
Thanks. Could you check it again?